hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1053,12 +1053,24 @@ public AbfsRestOperation read(final String path,
     return op;
   }
 
-  public AbfsRestOperation deletePath(final String path, final boolean recursive, final String continuation,
+  public AbfsRestOperation deletePath(final String path, final boolean recursive,
+                                      final String continuation,
                                       TracingContext tracingContext)
           throws AzureBlobFileSystemException {
     final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
-
     final AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
+
+    if (abfsConfiguration.isPaginatedDeleteEnabled() && recursive) {
```

Comment:
lets have a package-protected method `getIsPaginated()` for `abfsConfiguration.isPaginatedDeleteEnabled()` which can be spied in the test. This would remove the need of set method in abfsConfiguration.

Reply:
Great you have added private method `isPaginatedDeleteEnabled` which computes the conditions. What I am proposing is that there be a package-protected method `getIsPaginatedDeleteEnabled()` which you can refactor with `abfsConfiguration.isPaginatedDeleteEnabled()`. Something like:

```
private Boolean isPaginatedDeleteEnabled(TracingContext tracingContext,
      boolean isRecursiveDelete) throws AzureBlobFileSystemException {
    return getIsPaginatedDeleteEnabled()
        && getIsNamespaceEnabled(tracingContext) && isRecursiveDelete;
  }

...

Boolean getIsPaginatedDeleteEnabled() {
   return abfsConfiguration.isPaginatedDeleteEnabled();
}
```